src/Corax/IndexWriter.cs

```diff
@@ -1677,7 +1679,11 @@ public void Commit<TStatsScope>(TStatsScope stats)
                     continue;
 
                 using (staticFieldScope.For(CommitOperation.TextualValues))
-                    InsertTextualField(entriesToTermsTree, indexedField, workingBuffer, ref keys);
+                {
+                    var inserter = new TextualFieldInserter(this, entriesToTermsTree, indexedField, workingBuffer);
+                    inserter.InsertTextualField();
+                    inserter.Dispose();
```

Comment:
Should we wrap `inserter` in `using`? What it `InsertTextualField();` will throw

Reply:
I see that during the dispose, apart from returning the buffers to the pool we call `llt.ReleaseCompactKey(ref k);` as well. Isn't it mandatory?

Reply:
Not really, `ReleaseCompactKey` is just a performance optimization (avoid allocations).

Reply:
Fixed, anyway